internal/driver/driver.go

```diff
@@ -0,0 +1,646 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	sdkModels "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	"github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+
+	v4l2device "github.com/vladimirvivien/go4vl/v4l2/device"
+	"github.com/xfrr/goffmpeg/transcoder"
+)
+
+var driver *Driver
+var once sync.Once
+
+type Driver struct {
+	ds            *service.DeviceService
+	lc            logger.LoggingClient
+	wg            *sync.WaitGroup
+	asyncCh       chan<- *sdkModels.AsyncValues
+	deviceCh      chan<- []sdkModels.DiscoveredDevice
+	activeDevices map[string]*Device
+	rtspHostName  string
+	rtspTcpPort   string
+	mutex         sync.Mutex
+}
+
+// NewProtocolDriver initializes the singleton Driver and returns it to the caller
+func NewProtocolDriver() *Driver {
+	once.Do(func() {
+		driver = new(Driver)
+	})
+	return driver
+}
+
+// Initialize performs protocol-specific initialization for the device service.
+func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModels.AsyncValues,
+	deviceCh chan<- []sdkModels.DiscoveredDevice) error {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	d.lc = lc
+	d.asyncCh = asyncCh
+	d.deviceCh = deviceCh
+	d.ds = service.RunningService()
+	d.activeDevices = make(map[string]*Device)
+	d.wg = new(sync.WaitGroup)
+
+	h := NewHttpHandler(d)
+	if err := d.ds.AddRoute(common.ApiBase+ApiRefreshDevicePaths, h.RefreshExistingDevicePaths, http.MethodPost); err != nil {
+		return fmt.Errorf("failed to add API route %s, error: %s", ApiRefreshDevicePaths, err.Error())
+	}
+
+	rtspServerHostName, ok := service.DriverConfigs()[RtspServerHostName]
+	if !ok {
+		rtspServerHostName = DefaultRtspServerHostName
+		d.lc.Warnf("service config %s not found. Use the default value: %s", RtspServerHostName, DefaultRtspServerHostName)
+	}
+	d.lc.Debugf("RTSP server hostname: %s", rtspServerHostName)
```

Comment:
Use Info since it is on start up
```suggestion
	d.lc.Infof("RTSP server hostname: %s", rtspServerHostName)
```

Reply:
Changed severity level to Info. https://github.com/edgexfoundry-holding/device-usb-camera/pull/11/commits/f947e30ba36bca58cf452328e83108c19d23a6f3